quinn-proto/src/connection/packet_builder.rs

```diff
@@ -128,13 +128,16 @@ impl PacketBuilder {
             unreachable!("tried to send {:?} packet without keys", space_id);
         };
 
-        // Each packet must be large enough for header protection sampling, i.e. the
-        // combined lengths of the encoded packet number and protected payload must be at
-        // least 4 bytes longer than the sample required for header protection
+        // Each packet must be large enough for header protection sampling, i.e. the combined
+        // lengths of the encoded packet number and protected payload must be at least 4 bytes
+        // longer than the sample required for header protection. Further, each packet should be at
+        // least tag_len + 6 bytes larger than the destination CID on incoming packets so that the
+        // peer may send stateless resets that are indistinguishable from regular traffic.
 
         // pn_len + payload_len + tag_len >= sample_size + 4
         // payload_len >= sample_size + 4 - pn_len - tag_len
-        let min_size = buffer.len() + (sample_size + 4).saturating_sub(number.len() + tag_len);
+        let min_size = (buffer.len() + (sample_size + 4).saturating_sub(number.len() + tag_len))
```

Comment:
Style nit: I usually find it hard to make sense of asymmetric `.max()` in my head -- how would you feel about making it `Ord::max(.., ..)` instead? (`Ord` is in the prelude.)

Reply:
I like `.max` when chaining more than two, but I'm fine with switching to the qualified version here. Done.